test/deprecates.jl

```diff
@@ -0,0 +1,201 @@
+using StatsBase
+using Base.Test
+using Compat
+import Compat: view
+
+@testset "StatsBase.Deprecates" begin
+
+@testset "Deprecates WeightVec and weights" begin
```

Comment:
Would it make sense to keep the `ENV["TEST_DEPRECATES"]` behaviour around to make testing locally easier?